src/main/kotlin/com/meltwater/docker/compose/DockerCompose.kt

```diff
@@ -11,61 +12,108 @@ import org.slf4j.LoggerFactory
+        }
 
-        if (!File(ymlTmpfilename).exists()) {
-            throw RuntimeException("File $ymlTmpfilename could not be found.")
+        private val STDOUT_LOG_CONSUMER = { line: String ->
+            LOGGER.debug("OUT: $line")
         }
 
-        verifyDockerComposeInstallation()
-    }
+        private val STDERR_LOG_CONSUMER = { line: String ->
+            LOGGER.info("ERR: $line")
+        }
 
-    @Suppress("UNCHECKED_CAST")
-    private fun saveResourceToTmpFolder(classPathYmlResource: String): String {
-        try {
-            val ymlStream: InputStream = DockerCompose::class.java.classLoader.getResourceAsStream(classPathYmlResource)
-            val tmpFile = File.createTempFile(prefix, ".yml")
-            tmpFile.writeBytes(IOUtils.toString(ymlStream, Charset.forName("utf-8")).toByteArray())
-            ymlStream.close()
-            return tmpFile.path
-        } catch (ex: Exception) {
-            throw RuntimeException("Could not load yml file from classpath: $classPathYmlResource", ex)
+        private var MIN_DOCKER_COMPOSE_VERSION: DefaultArtifactVersion = DefaultArtifactVersion("1.5.0")
+
+
+        private const val YAML_FILE_EXTENSION = ".yml"
+
+        private const val ENVIRONMENT_FILE_EXTENSION = ".env"
+
+        private val ENVIRONMENT_VARIABLE_SEPARATOR = Pattern.compile("=")
+
+        private fun initializeNewDockerComposeSession(classPathYmlResource: String, env: Map<String, String>): String {
+            val composeFile = File.createTempFile(classPathYmlResource, YAML_FILE_EXTENSION)
+            val dockerComposeFilename = composeFile.absolutePath
+            saveResourceToTmpFolder(classPathYmlResource, composeFile)
+            if (!composeFile.exists()) {
+                throw RuntimeException("File $dockerComposeFilename could not be found.")
+            }
+            val environmentFileName = dockerComposeFilename.substringBeforeLast(YAML_FILE_EXTENSION) + ENVIRONMENT_FILE_EXTENSION
+            saveEnvironmentFile(env, environmentFileName)
+            return dockerComposeFilename
+        }
+
+        private fun saveResourceToTmpFolder(classPathYmlResource: String, destinationFile: File) {
+            try {
+                val ymlStream: InputStream = DockerCompose::class.java.classLoader.getResourceAsStream(classPathYmlResource)
+                val yamlContents = IOUtils.toString(ymlStream, Charset.forName("utf-8"))
+                destinationFile.writeBytes(yamlContents.toByteArray())
+                ymlStream.close()
+            } catch (ex: Exception) {
+                throw RuntimeException("Could not load yml file from classpath: $classPathYmlResource", ex)
+            }
         }
+
+        private fun saveEnvironmentFile(env: Map<String, String>, environmentFileName: String) {
+            val environmentPath = Paths.get(environmentFileName)
+            val environment = env.entries.stream().map { "${it.key}=${it.value}" }.collect(Collectors.joining("\n"))
```

Comment:
This can be done with kotlin map operators instead of java streams.

Try:
env.entries.joinToString("\n") { "${it.key}=${it.value}" }

If you see yourself using java streams in kotlin code you are doing something wrong :-)

Reply:
Will do :+1:

Reply:
Done